NewForce-Cohort-6/tabloidfullstack-cranberry-sauce
# Jg search posts by tag 42

# Description
[Ticket number 42](https://trello.com/c/POUT9Mb4/42-42-search-by-tag) added the  functionality for viewing posts by tag. For the time being I’ve put this in post management. 

# Testing Instructions

1.  `git add` and `git commit` whatever you're working on on your branch
2.  `git checkout main`
3. ` git fetch --a`
4.  `git checkout jg-ListPostsByTag-42` to switch to my branch.
5. If you don't see the most recent version of the files, try` git pull origin jg-ListPostsByTag-42`.
6. If you haven't created the Tabloid FullStack database, please do so.
7.  Use this to fill out tables and columns: [SQL data](https://github.com/NewForce-Cohort-6/tabloidfullstack-cranberry-sauce/blob/main/SQL/01_Tabloid_Create_DB.sql)
8.  Use this to seed the database: [SQL Seed Data](https://github.com/NewForce-Cohort-6/tabloidfullstack-cranberry-sauce/blob/main/SQL/02_Tabloid_Seed_Data.sql)
9.  The PostTag table is not populated in the seed data. Do an SQL query on your Tags table to see what ids each tag has: SELECT * FROM Tag
10. After you identify what ids are on the available tags write an insert query that populates the PostTag table with several different tags on a few posts, for example:   INSERT INTO PostTag VALUES (1, 3), (1, 1), (2, 4), (3,2), (3,4);  Add as many PostTag pairings as you like to test the feature.	
11. Execute Tabloid from inside of Visual Studio by clicking on the solid green arrow. This will launch Swagger and run the back end of the application to provide the API that will be utilized to view the front end. 
12. From your terminal run npm start from the `client` directory within `tabloidfullstack-cranberry-sauce`
13.  Step 12 will bring up the front end of the application in a browser window titled React App. In the url of that browser page, please Login using an email that you know is there and works for you (example: foo@bar.com from seed data)
14. Navigate to the PostManagement page using the nav bar.  You should see a list of all approved Post printed down the page. 
15. At the top of the page is a dropdown select menu labeled ‘List Posts by Tag’
16. Select a tag from that menu. If you select one that you populated with an PostTag association in the database, you should see the titles only of the posts associated with that tag. 17. There is a button next to each title to go to the post, but for now this only takes you to a page with the message “Not yet implemented”.
18. Toggle the tag selection between a few different tags to observe the behavior of the page. Returning the selection the the base “Select a Tag” returns all the posts to the page.

# Checklist:
